plugins/tiddlywiki/markdown/wrapper.js

```diff
@@ -3,338 +3,144 @@ title: $:/plugins/tiddlywiki/markdown/wrapper.js
-				(i+1) >= remarkableTree.length ||
-				!accumulatingTypes[remarkableTree[i+1].type]
-			)
-		) {
-			// The Markdown compiler thinks this is just text.
-			// Hand off to the WikiText parser to see if there's more to render
-			// But only if it's configured to, and we have more than whitespace
-			if(!pluginOpts.renderWikiText || accumulatedText.match(/^\s*$/)) {
-				out.push({
-					type: "text",
-					text: accumulatedText
-				});
-			} else {
-				// If we're inside a block element (div, p, td, h1), and this is the first child in the tree,
-				// handle as a block-level parse. Otherwise not.
-				var parseAsInline = !(isStartOfInline && i === 0);
-				var textToParse = accumulatedText;
-				if(pluginOpts.renderWikiTextPragma !== "") {
-					textToParse = pluginOpts.renderWikiTextPragma + "\n" + textToParse;
-				}
-				var wikiParser = $tw.wiki.parseText("text/vnd.tiddlywiki", textToParse, {
-					parseAsInline: parseAsInline
-				});
-				var rs = wikiParser.tree;
+	//console.log(JSON.stringify(mdTree,null,2));
+	//console.log((pluginOpts.renderWikiText ? "RenderWiki ON:" : "RenderWki OFF:") + "\n----------------\n" + textToParse);
 
-				// If we parsed as a block, but the root element the WikiText parser gave is a paragraph,
-				// we should discard the paragraph, since the way Remarkable nests its nodes, this "inline"
-				// node is always inside something else that's a block-level element
-				if(!parseAsInline
-					&& rs.length === 1
-					&& rs[0].type === "element"
-					&& rs[0].tag === "p"
-				) {
-					rs = rs[0].children;
-				}
+	var wikiParser = $tw.wiki.parseText("text/vnd.tiddlywiki",textToParse,{parseAsInline: false, wiki: options.wiki});
 
-				// If the original text element started with a space, add it back in
-				if(rs.length > 0
-					&& rs[0].type === "text"
-					&& (accumulatedText[0] === " " || accumulatedText[0] === "\n")
-				) {
-					rs[0].text = " " + rs[0].text;
-				}
-				out = out.concat(rs);
+	if(wikiParser.tree.length > 0) {
+		var hasWikiLinkRule = false;
+		// see if wikilink rule has been invoked
+		$tw.utils.each(wikiParser.inlineRules,function(ruleInfo) {
+			if(ruleInfo.rule.name === "wikilink") {
+				hasWikiLinkRule = true;
+				return false;
 			}
-			accumulatedText = '';
+		});
+		if(hasWikiLinkRule) {
+			unWikiLinkCode(wikiParser.tree);
 		}
+		decodeAttributes(wikiParser.tree);
 	}
-	return out;
-}
-
-var MarkdownParser = function(type, text, options) {
-	var tree = md.parse(text, {});
-	//console.debug(tree);
-	tree = convertNodes(tree);
-	//console.debug(tree);
-
-	this.tree = tree;
+	this.tree = wikiParser.tree;
+	this.source = text;
+	this.type = "text/x-markdown";
+	this.wiki = options.wiki;
 };
 
-exports["text/x-markdown"] = MarkdownParser;
+// to extend MarkdownIt outside of this module, do:
+//
+// md = $tw.Wiki.parsers["text/x-markdown"].prototype.md;
+// md.use(plugin[, options]);
+MarkdownParser.prototype.md = md;
 
-})();
+exports["text/x-markdown"] = MarkdownParser;
```

Comment:
`text/markdown` is an official MIME type now. So it needs to export both, the old an the new

Reply:
Thanks @pmario. Done! 

`text/markdown` is now the default markdown type, but `text/x-markdown` remains recognized.